bot/exts/info/site.py

```diff
@@ -122,20 +121,27 @@ async def site_rules(self, ctx: Context, *rules: int) -> None:
             return
 
         full_rules = await self.bot.api_client.get('rules', params={'link_format': 'md'})
-        invalid_indices = tuple(
-            pick
-            for pick in rules
-            if pick < 1 or pick > len(full_rules)
+
+        # Remove duplicates and sort the invalid rule indices
+        invalid_indices = sorted(
+            set(
+                pick
+                for pick in rules
+                if pick < 1 or pick > len(full_rules)
+            )
         )
 
         if invalid_indices:
-            indices = ', '.join(map(str, invalid_indices))
+            indices = ', '.join(str(index) for index in invalid_indices)
             await ctx.send(f":x: Invalid rule indices: {indices}")
             return
 
         for rule in rules:
             self.bot.stats.incr(f"rule_uses.{rule}")
 
+        # Remove duplicates and sort the rule indices
+        rules = sorted(set(rules))
```

Comment:
Do this before getting invalid indices to avoid having to iterate through duplicates during the search and avoid sorting again. Example

```py
rules = sorted(set(rules))
invalid = ', '.join(str(i) for i in rules if i < 1 or i > len(full_rules))

if invalid:
    ...
```

Reply:
Good catch, addressed in 2553a1d